src/ansys/fluent/core/utils/data_transfer.py

```diff
@@ -110,15 +109,15 @@ def transfer_case(
             container_workdir = PurePosixPath(container_workdir)
     for idx in range(num_files_to_try):
         file_name_tmp = (file_name_stem or "temp_case_file") + "_" + str(idx)
-        folder = Path(tempfile.mkdtemp(prefix="temp_store-", dir=workdir))
-        file_name = folder / file_name_tmp
         if inside_container:
-            file_name_container = container_workdir / folder.name / file_name_tmp
-            network_logger.debug(f"file_name: {file_name}")
+            file_name_container = container_workdir / file_name_tmp
+            network_logger.debug(f"file_name: {file_name_tmp}")
```

Comment:
I don't agree with marking https://github.com/ansys/pyfluent/pull/2583#discussion_r1549796105 as resolved. 

In my understanding, the fact that we had to move the files from a deeper directory to the root directory as done in these changes, in order for the file transfer service to work, exposes a flaw in the file transfer service: it does not properly support nested folders and files. 

Is there really nothing that we can do about this? Are we saying that users of this service would have to keep all their files in the root working directory and will be unable to work with nested directories? Seems like a major flaw to me.

Reply:
I already discussed the folder part with the team therefore that issue has resolved now. I already explained working, usage of code in previous comments and another point is that we are not the owner of code that we are using for upload and download service. We are user of both PIM and gRPC file transfer service packages. The expected functionality mentioned in above comment is not provided by the original authors and that is also not the requirement for our code. For concern about specification of remote file directory the point of contact should be PIM and gRPC file transfer team.

I already explained that gRPC FTS is being used in context of Docker container where we have 2 containers running remotely and we are mapping their host and container working dir.

Fluent 
host_path = pyfluent.EXAMPLES_PATH
work_dir = /mnt/pyfluent

gRPC
host_path =  pyfluent.EXAMPLES_PATH
work_dir = /home/container/workdir

gRPC FTS will upload file to `/home/container/workdir` and it will be available in `pyfluent.EXAMPLES_PATH`

Fluent will read file from `pyfluent.EXAMPLES_PATH`.

Fluent will write file to `/mnt/pyfluent` and it will be available in `pyfluent.EXAMPLES_PATH`

gRPC FTS will download file from  `pyfluent.EXAMPLES_PATH` to  specified directory.

`pyfluent.EXAMPLES_PATH` is fixed for all scripts that are being used in docker.

We want to upload files to already defined directory and not in nested directory. Therefore the point of uploading file into remote nested folder is not the requirement here. 

If we specify remote file like `pyfluent.EXAMPLES_PATH/random_folder/file_name`. then Fluent will not be able to read it when it runs in container mode. This was the issue in data_transfer module and we have resolved it by removing folder part from code. Now we can see that all unit tests have passed for 23R1. 

https://github.com/ansys/pyfluent/actions/runs/8633461784/job/23672130550?pr=2583

Therefore I have marked all discussions related to removal of folder part as resolved. 

Thank you.

Reply:
Thank you for the detailed explanation. I'm probably just being dense but I'm still confused. It still looks to me like it should be possible to use relative paths and support nested files and directories. 

Anyway that is fine, I don't want to block this if I'm the only one with an issue. I don't have the time right now to properly investigate this, I will look into this later and let you know if I still have any concerns.